src/settingsmanager.cpp

```diff
@@ -13,12 +13,18 @@ SettingsManager::SettingsManager(QObject *parent)
     initSettings();
 }
 
-SettingsManagerView* SettingsManager::getView()
+SettingsView* SettingsManager::getView()
 {
-    auto view = new SettingsManagerView();
-    view->registerObject("settingsManager", this);
-    view->setHtml();
-    connect(view, &QObject::destroyed, this, [=]() { m_settingsViewDisplayed = false; });
+    static SettingsView *view = nullptr;
+
+    if (!view || !m_settingsViewDisplayed)
+        view = new SettingsView();
+
+    view->init(m_kiwixServerPort, m_zoomFactor * 100, m_downloadDir);
+    connect(view, &QObject::destroyed, this, [=]() { m_settingsViewDisplayed = false;});
+    connect(view, &SettingsView::serverPortChanged, this, &SettingsManager::setKiwixServerPort);
+    connect(view, &SettingsView::downloadDirChanged, this, &SettingsManager::setDownloadDir);
+    connect(view, &SettingsView::zoomFactorChanged, this, &SettingsManager::setZoomFactor);
     m_settingsViewDisplayed = true;
     return view;
```

Comment:
This method is globally wrong.
- It is surprising to have a static variable here. It would be more "natural" to use a member of `SettingManager`
- If we have not null `view` and false `m_settingViewDisplayed`, it means that the view has been destroyed. So `view` is a dandling pointer. This is a "wrong" situation.
- If we call `getView` when the view is displayed, we will init and connect twice.
- `m_settingViewDisplayed` is never used, we could remove it (it was already the case before, but let's simplify this code in this PR)
- In fact, we don't need to keep a reference on `view` here as Qt will handle the lifetime of the widget for us once we will add it to the parent widget. (And we could rename this method to `createView`)